crates/turborepo-telemetry/src/events/mod.rs

```diff
@@ -41,3 +41,19 @@ pub trait EventBuilder {
     fn track(&self, event: Event);
     fn child(&self) -> Self;
 }
+
+#[macro_export]
+macro_rules! track_usage {
+    ($tel:expr, $field:expr, $is_used:expr) => {
+        if $is_used($field) {
+            $tel.track_arg_usage(
+                stringify!($field)
+                    .trim_start_matches("&")
+                    .trim_start_matches("self.")
```

Comment:
Open to better options here if you have suggestions @chris-olszewski !

Reply:
I agree this feels gross, played around with this and couldn't come up with anything better. What we really want is proc macro. Maybe I'll take a late DYF and try to whip something up.